authentication/java/azure/lib/src/main/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProvider.java

```diff
@@ -33,9 +35,20 @@ public AzureIdentityAccessTokenProvider(@Nonnull final TokenCredential tokenCred
         } else {
             _scopes = Arrays.asList(scopes);
         }
+        if (allowedHosts == null || allowedHosts.length == 0) {
```

Comment:
Maintaining the Host Validation logic in the BaseBearerTokenAuthProvider will help in code reuse. 

```
class BaseBearerTokenAuthenticationProvider{
  public authRequest(){
   if(isHostValid){
    const token = getAuthToken();
    }
  }

  public isHostValid: boolean{
  }
}

class CustomAuthenticationProvider extends BaseBearerTokenProvider(){
 public authRequest(){
   if(super.isHostValid()){
     const token = customAccessTokenProvider.getAuthToken();
    }
  }
}
```

Reply:
But it won't make it reusable for other authentication schemes/implementations.